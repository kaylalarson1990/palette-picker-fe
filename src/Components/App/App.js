import React, { Component } from "react";
import Generator from "../Generator/Generator";
import ProjectContainer from "../ProjectContainer/ProjectContainer"
import "./App.css";
import { fetchAllProjects, fetchAllPalettes } from "../../apiCalls";

export class App extends Component {
  constructor() {
    super();
    this.state = {
      projects: [],
      palettes: []
    };
  }

  componentDidMount() {
    fetchAllProjects().then(data => this.setState({ projects: data }));
    fetchAllPalettes().then(data => this.setState({ palettes: data }));
  }

  render() {
    return (
      <div className="App">
        <Generator />
        <ProjectContainer />
      </div>
    );
  }
}

// export const mapDispatchToProps = dispatch => ({
//   fetchAllProjects: () => dispatch(fetchAllProjects()),
//   fetchAllPalettes: () => dispatch(fetchAllPalettes())
// });

export default App;
