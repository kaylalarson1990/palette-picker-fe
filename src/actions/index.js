export const gatherProjects = projects => ({
  type: 'GATHER_PROJECTS',
  projects
})

export const gatherPalettes = palettes => ({
  type: 'GATHER_PALETTES',
  palettes
})

export const gatherProject = id => ({
  type: 'GATHER_PROJECT',
  id
})

export const gatherPalette = id => ({
  type: 'GATHER_PALETTE',
  id
})