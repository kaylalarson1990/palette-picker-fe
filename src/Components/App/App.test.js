import React from 'react';
import { App, mapDispatchToProps } from './App';
import { shallow } from "enzyme";
import { gatherPalettes } from '../../actions/index'
import "../../../setupTests";

describe("App", () => {
  let wrapper, instance, mockFunc;

  beforeEach(() => {
    wrapper = shallow(<App />);
    instance = wrapper.instance();
    mockFunc = jest.fn();
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
})

describe("mapDispatchToProps", () => {
  it("calls dispatch with a gatherPalettes action", () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = gatherPalettes("Test Palette");
    const mappedDispatch = mapDispatchToProps(mockDispatch);
    mappedDispatch.gatherPalettes("Test Palette");
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});