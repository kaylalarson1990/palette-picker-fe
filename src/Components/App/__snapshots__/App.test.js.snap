// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="App"
>
  <Connect(Generator) />
  <Connect(ProjectContainer) />
</div>
`;
